operator/pkg/ingress/service.go

```diff
@@ -0,0 +1,240 @@
+		//doing nothing right now
+	default:
+		err = fmt.Errorf("received an unknown event: %s", ev)
+	}
+	return err
+}
+
+func (sm *serviceManager) handleServiceAddedEvent(event serviceAddedEvent) error {
+	sm.notify(event.service)
+	return nil
+}
+
+func (sm *serviceManager) handleServiceUpdatedEvent(event serviceUpdatedEvent) error {
+	if event.newService.ObjectMeta.GetDeletionTimestamp() != nil {
+		// This means the service is in the process of being deleted, cleaning up load balancers
+		// and such. Nothing to do in this case.
+		log.WithFields(logrus.Fields{
+			"namespace": event.newService.Namespace,
+			"name":      event.newService.Name,
+		}).Debug("Service is being deleted")
+		return nil
+	}
+	log.WithField("old", event.oldService).WithField("new", event.newService).Info("Handling service update")
+	sm.notify(event.newService)
+	return nil
+}
+
+// notify informs parent ingress about load balancer details
+func (sm *serviceManager) notify(service *slim_corev1.Service) {
+	if len(service.Status.LoadBalancer.Ingress) > 0 {
+		log.Info("Notify ingress controller for service ingress")
+		sm.ingressQueue.Add(ingressServiceUpdatedEvent{ingressService: service})
+	}
+}
+
+func getServiceForIngress(ingress *slim_networkingv1.Ingress) *v1.Service {
+	var targetPort int32
+	if len(ingress.Spec.Rules) > 0 && len(ingress.Spec.Rules[0].HTTP.Paths) > 0 &&
+		ingress.Spec.Rules[0].HTTP.Paths[0].Backend.Service != nil {
+		targetPort = ingress.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port.Number
+	} else {
+		if ingress.Spec.DefaultBackend != nil && ingress.Spec.DefaultBackend.Service != nil {
+			targetPort = ingress.Spec.DefaultBackend.Service.Port.Number
+		}
+	}
+
+	ports := []v1.ServicePort{
+		{
+			Name:     "http",
+			Protocol: "TCP",
+			Port:     80,
+			// TODO(michi) how do we deal with multiple target ports?
+			// TODO(jarno) It seems that when service is accessed via
+			//             NodePort all these targetports are ignored anyway,
+			//             and the targetport defined in the backend service is
+			//             used regardless what is specified here.
+			TargetPort: intstr.FromInt((int)(targetPort)),
+		},
+	}
+	if len(ingress.Spec.TLS) > 0 {
+		ports = []v1.ServicePort{
+			{
+				Name:     "https",
+				Protocol: "TCP",
+				Port:     443,
+				// TODO(michi) how do we deal with multiple target ports?
+				TargetPort: intstr.FromInt((int)(ingress.Spec.Rules[0].HTTP.Paths[0].Backend.Service.Port.Number)),
+			},
+		}
+	}
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getServiceNameForIngress(ingress),
+			Namespace: ingress.Namespace,
+			Labels:    map[string]string{ciliumIngressLabelKey: "true"},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         slim_networkingv1.SchemeGroupVersion.String(),
+					Kind:               "Ingress",
+					Name:               ingress.Name,
+					UID:                ingress.UID,
+					BlockOwnerDeletion: boolP(true),
```

Comment:
we can drop this, see https://github.com/cilium/cilium/pull/19074/commits/37bcb4ddce8dac8776c496bfb7c1dab44cde0a57